client/src/components/app/exam/TopbarTimer.vue

```diff
@@ -94,17 +103,38 @@ export default {
 
   methods: {
     startTimer() {
-      this.timer = setInterval(this.updateTimer, 1000)
+      this.updateTimer()
+      this.timer = setInterval(this.updateTimer, 50) // 50 ms is minimum guarenteed browser refresh rate
     },
 
     updateTimer() {
       const currentTimeSeconds = Math.floor(Date.now() / 1000)
       const elapsedSeconds = currentTimeSeconds - this.startEpoch!
-      this.secondsRemaining = this.duration.valueOf() * 60 - elapsedSeconds
+      this.secondsRemaining = this.quizDuration.valueOf() * 60 - elapsedSeconds
 
       if (this.secondsRemaining <= 0) {
         this.secondsRemaining = 0
         this.stopTimer()
+
+        // submit on timeout
+        const mutation = this.$apollo.mutate({
+          mutation: SubmitUserQuizQuestionsMutation,
+          variables: {
+            input: {
+              userQuizID: this.$route.params.quizID
+            }
+          }
+        })
+        this.examStore.submitting = true
```

Comment:
you need to set this flag back to 'false' or it will always say submitting. The other fix we could do is on load of the exams page we reset this flag